test/protocol/protocol.test.js

```diff
@@ -0,0 +1,648 @@
+                blockchain: awilix.asClass(Blockchain).singleton(),
+                network: awilix.asClass(MockNetwork).singleton(),
+                graphStorage: awilix.asValue(new MockGraphStorage()),
+                challenger: awilix.asValue({ startChallennodeWeb3ging: () => { log.info('start challenging.'); } }),
+                logger: awilix.asValue(log),
+                dcService: awilix.asClass(DCService),
+                remoteControl: awilix.asClass(MockRemoteControl),
+            });
+
+            const blockchain = container.resolve('blockchain');
+            const dcService = container.resolve('dcService');
+
+            testNode.container = container;
+        });
+
+        // Let only first node listen to events.
+        // TODO
+        // Set event listener
+        function listenBlockchainEvents(blockchain) {
+            const delay = 500;
+            let working = false;
+            let deadline = Date.now();
+            return setInterval(() => {
+                if (!working && Date.now() > deadline) {
+                    working = true;
+                    blockchain.getAllPastEvents('BIDDING_CONTRACT');
+                    blockchain.getAllPastEvents('READING_CONTRACT');
+                    blockchain.getAllPastEvents('ESCROW_CONTRACT');
+                    deadline = Date.now() + delay;
+                    working = false;
+                }
+            }, 500);
+        }
+
+        testNode1.eventListener = listenBlockchainEvents(testNode1.blockchain);
+    });
+
+    afterEach('Unregister container', async () => {
+        testNodes.forEach((testNode) => {
+            if (testNode.container) {
+                testNode.container.dispose(); // Promise.
+                testNode.container = undefined;
+            }
+
+            if (testNode.eventListener) {
+                clearInterval(testNode.eventListener);
+            }
+        });
+    });
+
+    it('should successfully create profile', async function createProfile() {
+        this.timeout(10000);
+
+        let profileInfo = await testNode1.blockchain.getProfile(testNode1.wallet);
+        expect(profileInfo.active).to.be.false;
+
+        await testNode1.blockchain.createProfile(testNode1.identity, 2, '1', '1', '100000');
+
+        profileInfo = await testNode1.blockchain.getProfile(testNode1.wallet);
+        expect(profileInfo.active).to.be.true;
+
+        const events = await biddingInstance.getPastEvents('allEvents', {
+            fromBlock: 0,
+            toBlock: 'latest',
+        });
+
+        expect(events).to.have.lengthOf(1);
+        expect(events[0].event).to.equal('ProfileCreated');
+        expect(events[0].returnValues).to.have.property('wallet').that.deep.equals(testNode1.wallet);
+        expect(events[0].returnValues).to.have.property('node_id').that.deep.equals(testNode1.getIdentityExtended());
+    });
+
+    describe('DC replication', () => {
+        const vertices = [
```

Comment:
Can we separate test data into seperate file, so it can be potentially reused in further tests plus this code becomes more compact and readable please?

Reply:
Agree. Currently they're here for the demo purpose only. Same logic I'd apply for all mocking classes.